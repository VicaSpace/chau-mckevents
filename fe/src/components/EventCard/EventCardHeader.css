.card-head {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    padding: 0px 24px;

    width: 832.95px;
    height: 56px;

    background: #FFFFFF;
}


.event-name {
    width: 148px;
    height: 24px;

    margin-top: 16px;
    margin-left: 24px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;

    color: rgba(0, 0, 0, 0.85);
}


.event-organizer {
    position: absolute;
    width: 247px;
    height: 20px;
    margin-left: 254px;
    margin-top: 36px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 19px;

    color: #000000;
    
}

.event-office {
    position: absolute;
    width: 117px;
    height: 22px;

    margin-left: 691.95px;
    margin-top: 17px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;

    color: #1890FF;
}