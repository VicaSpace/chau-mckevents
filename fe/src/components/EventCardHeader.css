.card-head {
    /* Auto layout */
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 0px 24px;

    width: 832.95px;
    height: 56px;

    /* Neutral/1 */

    background: #FFFFFF;

    /* Inside auto layout */

    flex: none;
    order: 0;
    align-self: stretch;
    flex-grow: 0;
}


.event-name {
    width: 148px;
    height: 24px;

    margin-top: 16px;
    margin-left: 24px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    /* identical to box height, or 150% */

    /* Character/Primary .85 */

    color: rgba(0, 0, 0, 0.85);
}


.event-organizer {
    position: absolute;
    width: 247px;
    height: 20px;
    margin-left: 254px;
    margin-top: 36px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 19px;

    color: #000000;
}

.event-office {
    width: 117px;
    height: 22px;
    margin-left: 691.95px;

    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;
    /* identical to box height, or 157% */


    /* Primary/6 */

    color: #1890FF;


    /* Inside auto layout */

    flex: none;
    order: 0;
    flex-grow: 0;
}