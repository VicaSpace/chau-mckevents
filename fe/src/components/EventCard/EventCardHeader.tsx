import React from 'react';
import './EventCardHeader.css';

const EventCardHeader: React.FC<{
  name: string;
  organizer: string;
  office: string;
}> = ({ name, organizer, office }) => {
  return (
    <div className="card-head" data-testid="test-card-head">
      <div className="event-name" data-testid="test-event-name">
        {name}
      </div>
      <div className="event-organizer" data-testid="test-event-organizer">
        Organizer: {organizer}
      </div>
      <div className="event-office" data-testid="test-event-office">
        {office}
      </div>
    </div>
  );
};

export default EventCardHeader;
