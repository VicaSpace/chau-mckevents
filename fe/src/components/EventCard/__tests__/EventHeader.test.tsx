import {render, screen} from '@testing-library/react'
import React from 'react'
import EventCardHeader from '../EventCardHeader'

const dummyProps = {
    name: 'test',
    organizer: 'test-organizer',
    office: 'test-office',
}

describe('Test: Event Card Header component screenshot', () => {
    it('should render the Event Card correctly', () => {
      const { container } = render(
          <EventCardHeader {...dummyProps} />
      );
      expect(container).toMatchSnapshot();
    });
  });

describe('Test: Event Card Header props', () => {
    it('should render name node ', () => {
        render(<EventCardHeader {...dummyProps} />);
        const nameNode = screen.getByTestId('test-event-name');
        expect(nameNode.textContent).toBe(dummyProps.name);
    });
    it('should render organizer node ', () => {
        render(<EventCardHeader {...dummyProps} />);
        const organizerNode = screen.getByTestId('test-event-organizer');
        expect(organizerNode.textContent).toBe(`Organizer: ${dummyProps.organizer}`);
    });
    it('should render office node ', () => {
        render(<EventCardHeader {...dummyProps} />);
        const officeNode = screen.getByTestId('test-event-office');
        expect(officeNode.textContent).toBe(dummyProps.office);
    });
});
