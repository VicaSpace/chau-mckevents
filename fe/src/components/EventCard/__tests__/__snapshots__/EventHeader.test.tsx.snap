// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Event Card Header component screenshot should render the Event Card correctly 1`] = `
<div>
  <div
    class="card-head"
    data-testid="test-card-head"
  >
    <div
      class="event-name"
      data-testid="test-event-name"
    >
      test
    </div>
    <div
      class="event-organizer"
      data-testid="test-event-organizer"
    >
      Organizer: 
      test-organizer
    </div>
    <div
      class="event-office"
      data-testid="test-event-office"
    >
      test-office
    </div>
  </div>
</div>
`;
